### PR TITLE
refactor: transaction decoding using the wrong field

### PR DESCRIPTION
**Pull request by fubuloubu**
fixes: #7

### CODE REVIEW

**Review by helloibis — APPROVED** [review]

**Comment by fubuloubu** [comment]
Need to test this with local network, I think there might be problems

**Comment by unparalleled-js** [comment]
> Need to test this with local network, I think there might be problems

I tested local networks on both main and this branch and I didn't detect any differences. However, reverts seem to fail right now with an integer overload error. I will create an issue for this.

I am also in the works of testing out the actual issue.

File: ape_optimism/ecosystem.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def create_transaction(self, **kwargs) -> TransactionAPI:
             kwargs["chainId"] = int(kwargs["chainId"], 16)
 
         if "input" in kwargs:
```
Comment (unparalleled-js) [inline ape_optimism/ecosystem.py:81]: Does this `if` statement still make sense?

**Review by unparalleled-js — DISMISSED** [review]
I tried to deploy a contract on Optimism Goerli and it errored:

```
     14 def decode_hex(value: str) -> bytes:
     15     if not is_text(value):
---> 16         raise TypeError("Value must be an instance of str")
     17     non_prefixed = remove_0x_prefix(HexStr(value))
     18     # unhexlify will only accept bytes type someday

TypeError: Value must be an instance of str
```

Edit: fixed!

File: ape_optimism/ecosystem.py [hunk 2]
```diff
@@ -78,11 +78,11 @@ def create_transaction(self, **kwargs) -> TransactionAPI:
         if isinstance(kwargs.get("chainId"), str):
             kwargs["chainId"] = int(kwargs["chainId"], 16)
 
-        if "input" in kwargs:
-            kwargs["data"] = decode_hex(kwargs.pop("hash"))
+        if "hash" in kwargs:
```
Comment (fubuloubu) [inline ape_optimism/ecosystem.py:81]: ahhhh, whoops

**Review by fubuloubu — COMMENTED** [review]
Looks great, thanks @unparalleled-js!

**Comment by fubuloubu** [comment]
Weird failure on previous CI pass

**Comment by unparalleled-js** [comment]
> Weird failure on previous CI pass

could it be the auto-cancel? If you push before the last one completes, it cancels the last ones now.

**Pull request merged**